### PR TITLE
Convert some tests to javascript

### PR DESCRIPTION
**Pull request by Page-**
Change-type: patch

### CODE REVIEW

File: src/compose/network.ts [hunk 1]
```diff
@@ -94,7 +94,9 @@ export class Network {
 	public static fromComposeObject(
 		name: string,
 		appId: number,
-		network: Partial<ComposeNetworkConfig>,
+		network: Partial<Omit<ComposeNetworkConfig, 'ipam'>> & {
+			ipam?: Partial<ComposeNetworkConfig['ipam']>;
+		},
```
Comment (Page-) [inline src/compose/network.ts]: This is more accurate to what tests pass in and the code also handles the `ipam` properties being undefined/null

File: src/compose/ports.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ interface PortRange {
 export class PortMap {
 	private ports: PortRange;
 
-	private constructor(portStrOrObj: string | PortRange) {
+	public constructor(portStrOrObj: string | PortRange) {
```
Comment (Page-) [inline src/compose/ports.ts]: This constructor is used a lot by the tests and the easiest way to make them work was to make it public

File: test/16-ports.spec.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+	});
+
+	describe('toDockerOpts', function() {
+		it('should correctly generate docker options', () =>
+			expect(new PortMap('80').toDockerOpts()).to.deep.equal({
+				exposedPorts: {
+					'80/tcp': {},
+				},
+				portBindings: {
+					'80/tcp': [{ HostIp: '', HostPort: '80' }],
```
Comment (Page-) [inline test/16-ports.spec.js]: I had to convert all of these to strings to match the types - I'm assuming that's correct but just want to double check

File: test/17-config-utils.spec.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+TIMEOUT 30
+
+MENU TITLE Boot Options
+LABEL primary
+MENU LABEL primary Image
+LINUX /Image
+APPEND \${cbootargs} \${resin_kernel_root} ro rootwait isolcpus=3\
+`;
+
+				let readFileStub = stub(fs, 'readFile').resolves(text);
```
Comment (Page-) [inline test/17-config-utils.spec.js]: I've changed to assigning the stub in a bunch of places as it helps a lot with the typings when we then call sinon specific methods

File: test/18-startup.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Supervisor } from '../src/supervisor';
+import { expect } from './lib/chai-config';
+
+describe('Startup', () => {
+	it('should startup correctly', function() {
+		const supervisor = new Supervisor();
+		expect(supervisor.init()).to.not.throw;
+		// Cast as any to access private properties
+		const anySupervisor = supervisor as any;
```
Comment (Page-) [inline test/18-startup.ts]: I'm not a massive fan of this cast but it's probably preferable to making all the properties public

File: test/18-startup.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Supervisor } from '../src/supervisor';
```
Comment (Page-) [inline test/18-startup.ts]: There's 2 files with `18-`, not sure if that's a problem but it surprised me slightly

File: src/compose/ports.ts (hunk 2 above)
Comment (CameronDiver) [reply src/compose/ports.ts:33]: There's a good reason that this is private, and I don't want anybody to think that this is the way to use that throughout the codebase. I'd 100% prefer a `@ts-ignore` or cast to any.

File: src/compose/ports.ts [hunk 7]
```diff
@@ -18,7 +18,7 @@ export interface DockerPortOptions {
 	portBindings: PortBindings;
 }
 
-interface PortRange {
+export interface PortRange {
```
Comment (Page-) [inline src/compose/ports.ts:21]: I had to expose this to be able to force cast with it in the tests

File: src/compose/ports.ts (hunk 2 above)
Comment (Page-) [reply src/compose/ports.ts]: done
